client/quantum_serverless/core/provider.py

```diff
@@ -332,16 +335,127 @@ def get_compute_resources(self) -> List[ComputeResource]:
         return resources
 
     def create_compute_resource(self, resource) -> int:
-        """Create compute resource for provider."""
-        raise NotImplementedError
+        """
+        Create compute resource for provider.
+
+        First, create a compute_template based on the defined ComputeResource.
+        If the compute_template already exists (which shouldn't be the case,
+        exit silently. Otherwise, create the template. This template is
+        ephemeral and will be deleted when no longer needed (either the cluster
+        was created or we failed to create it and thus need to start over).
+
+        Then use that template to create a KubeRay Cluster. If the cluster
+        already exists, we exit silently. Users are only able to configure the
+        amount of cpu/memory and the number of worker replicas. The full spec
+        can be found in the KubeRay API spec under the "protoCluster"
+        definition:
+        https://github.com/ray-project/kuberay/blob/master/proto/swagger/cluster.swagger.json
+        """
+        compute_resource_name = self.name + "-template"
+
+        # Create template from resource
+        req = requests.get(
+            f"{self.host}/apis/v1alpha2/namespaces/{self.namespace}"
+            f"/compute_templates/{compute_resource_name}",
+            timeout=TIMEOUT,
+        )
+        if req.status_code == 200:
+            print(f"template name {compute_resource_name} already exists")
+            return 1
+
+        data = {
+            "name": compute_resource_name,
+            "namespace": self.namespace,
+            "cpu": resource.resources["cpu"],
+            "memory": resource.resources["memory"],
+        }
+        req = requests.post(
+            f"{self.host}/apis/v1alpha2/namespaces/{self.namespace}"
+            f"/compute_templates",
+            json=data,
+            timeout=TIMEOUT,
+        )
+        if req.status_code != 200:
+            req.raise_for_status()
+            return 1
+
+        # Create cluster from template
+        req = requests.get(
+            f"{self.host}/apis/v1alpha2/namespaces/{self.namespace}"
+            f"/clusters/{self.name}",
+            timeout=TIMEOUT,
+        )
+        if req.status_code == 200:
+            print(f"cluster {self.name} already exists")
+            delete_kuberay_template(self.host, self.namespace, compute_resource_name)
+            return 1
+
+        data = {
+            "name": self.name,
+            "namespace": self.namespace,
+            "user": "default",
+            "clusterSpec": {
+                "headGroupSpec": {
+                    "computeTemplate": compute_resource_name,
+                    "image": "rayproject/ray:2.2.0",
```

Comment:
I think this should be our image.

We can have default nightly one 3.9. Also it would be good to be able to pass it through `resource` argument, or even have `**kwags` for `create_compute_resource` method. Or being able to set it up through env variables.

Maybe it will be something like that:
- if passed as parameter -> use this values
- if not passed as parameter, but env variable exist -> use this env var
- if no env var and passed parameter -> use default

Reply:
Did not know about the qiskit image...

Reply:
https://hub.docker.com/r/qiskit/quantum-serverless-ray-node/tags